src/components/model/ModelStatusRollup.svelte

```diff
@@ -0,0 +1,202 @@
+  export let mode: Mode = 'full';
+  export let model:
+    | Pick<ModelSlim, 'refresh_activity_type_logs' | 'refresh_model_parameter_logs' | 'refresh_resource_type_logs'>
+    | undefined;
+  export let selectable: boolean = false;
+  export let showCompleteStatus: boolean = true;
+  export let selectedLog: 'activity' | 'parameter' | 'resource' | undefined = undefined;
+
+  const dispatch = createEventDispatcher<{
+    select: ModelLog | null;
+  }>();
+  const rollupTooltipMessages: Record<ModelStatus, string> = {
+    complete: 'Model extraction complete',
+    error: 'Model has extraction errors',
+    extracting: 'Extracting model',
+    none: '',
+  };
+  const activityLogTooltipMessages: Record<ModelStatus, string> = {
+    complete: 'Activity type extraction complete',
+    error: 'Activity type extraction has errors',
+    extracting: 'Extracting activity types',
+    none: '',
+  };
+  const parameterLogTooltipMessages: Record<ModelStatus, string> = {
+    complete: 'Model parameter extraction complete',
+    error: 'Model parameter extraction has errors',
+    extracting: 'Extracting model parameters',
+    none: '',
+  };
+  const resourceLogTooltipMessages: Record<ModelStatus, string> = {
+    complete: 'Resource type extraction complete',
+    error: 'Resource type extraction has errors',
+    extracting: 'Extracting resource types',
+    none: '',
+  };
+
+  let activityLog: ModelLog | null = null;
+  let activityLogStatus: ModelStatus = 'none';
+  let parameterLog: ModelLog | null = null;
+  let parameterLogStatus: ModelStatus = 'none';
+  let resourceLog: ModelLog | null = null;
+  let resourceLogStatus: ModelStatus = 'none';
+  let status: ModelStatus = 'none';
+
+  $: ({
+    modelStatus: status,
+    activityLog,
+    activityLogStatus,
+    parameterLog,
+    parameterLogStatus,
+    resourceLog,
+    resourceLogStatus,
+  } = getModelStatusRollup(model));
+
+  function selectActivityLog() {
+    selectedLog = 'activity';
+    dispatch('select', activityLog);
+  }
+
+  function selectParameterLog() {
+    selectedLog = 'parameter';
+    dispatch('select', parameterLog);
+  }
+
+  function selectResourceLog() {
+    selectedLog = 'resource';
+    dispatch('select', resourceLog);
+  }
+</script>
+
+<div class="model-status-container">
+  {#if mode === 'rollup' || mode === 'iconOnly'}
+    <div class="model-status-rollup" class:icon-only={mode === 'iconOnly'}>
+      <ModelStatusIcon {showCompleteStatus} {status} tooltipContent={status && rollupTooltipMessages[status]} />
+      {#if mode === 'rollup'}
+        {#if status === 'extracting'}
+          Extracting
+        {:else if status === 'complete' && showCompleteStatus}
+          Extracted
+        {:else if status === 'error'}
+          Errors extracting
+        {/if}
+      {/if}
+    </div>
+  {:else}
+    <div class="model-status-logs-container" class:horizontal={flow === 'horizontal'}>
+      <button
+        disabled={!selectable}
```

Comment:
Is `selectable` computed anywhere or always true? Would be good to disallow selection of a button if there is no error to view

Reply:
`selectable` is currently not computed. I discussed this with @lklyne and he was okay with keeping it interactable even if there weren't errors, much like clicking the `console` when there are no errors and being presented with an empty console. I'm open to discussing which is better though!

Reply:
That's fair, I do think there should be some sort of state below to indicate that there are no errors or something. My first reaction when clicking on the successful tab was that something was missing.

Reply:
Could say something like "Success" in a white box that takes up the same amount of room as the error would?